src/components/editor/AppController.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { browserHistory } from 'react-router';
+import FalcorController from "lib/falcor/FalcorController";
+import { setAppReady } from "lib/falcor/falcorUtils";
+
+export default class AppController extends FalcorController {
+  constructor(props) {
+    super(props);
+    this.safeSetState({
+      defaultValue: this.handleDefaultValue.bind(this)()
```

Comment:
I guess I'll try and seperate my replies to your comments to each issue so they can have separate comments.

the bind part shouldn't be redundant as I access this.props.location.pathname in the function (assuming we end up keeping the function).

As for what it does, for the server render it decides which value should be shown in the dropdown, if a URL is accessed directly to the article part it will show articles in the dropdown from the start. 

I'll address the use of my approach further down